uwblueprint/abtc
# feat: restyle calendar to match Figma

## Notion ticket link

[Download, restyle, and implement Calendar Component ](https://www.notion.so/uwblueprintexecs/Download-restyle-and-implement-Calendar-Component-7c7a4536365d42a6ad5bc135c6b8e7c7?pvs=4)



## Implementation description
* Download react-big-calendar, an open source calendar component
* Restyle calendar component using CSS and customizing props (reference: https://jquense.github.io/react-big-calendar/examples/iframe.html?id=props-full-prop-list--page&viewMode=story)
* Create dummy event data to show how shifts would be shown
* Create visible static calendar that can switch between monthly and weekly view, and filter by request type


## Steps to test
1. Navigate to http://localhost:3000/calendar
2. By default monthly view should load and all events should be shown
3. "Month" should be shown in select dropdown in top right
4. Click on a date (i.e. the number, like 18)
5. Weekly view for the week containing that day should load
6. "Week" should be shown in select dropdown in top right
7. If Filter by is not selected, all events should be shown
8. Filter by and select Kitchen, only kitchen request events should be shown
9. Filter by and select Site, only site request events should be shown
10. Switching to Month or Week should work as expected
11. Clicking next and back arrows should navigate to the next month and previous month respectively
<img width="1440" alt="Screen Shot 2024-03-28 at 8 44 32 PM" src="https://github.com/uwblueprint/abtc/assets/88808428/ca0a64e5-d982-400f-9eef-31e68f219e95">


## What should reviewers focus on?
* Calendar can switch between monthly view and weekly view
* Calendar can filter by request type
* Calendar has the same styling and designs as the calendar included in the designs


## Checklist
